test/evergreen.yml

```diff
@@ -2883,9 +2881,9 @@ tasks:
     exec_timeout_secs: 25200
     commands:
       - func: "get project"
-      - func: "compile wiredtiger"
+      - func: "compile wiredtiger address sanitizer"
         vars: 
-          posix_configure_flags: -DENABLE_STRICT=1 -DHAVE_DIAGNOSTIC=1 -DNON_BARRIER_DIAGNOSTIC_YIELDS=1 -DHAVE_BUILTIN_EXTENSION_LZ4=1 -DHAVE_BUILTIN_EXTENSION_SNAPPY=1 -DHAVE_BUILTIN_EXTENSION_ZLIB=1 -DHAVE_BUILTIN_EXTENSION_ZSTD=1
+          NON_BARRIER_DIAGNOSTIC_YIELDS: -DNON_BARRIER_DIAGNOSTIC_YIELDS=1
```

Comment:
I am not a fan of having such a specific field. Can we have instead something like `extra_flags`?
```
extra_flags: -DNON_BARRIER_DIAGNOSTIC_YIELDS=1
```
And then we would use it wherever needed:
```
{extra_flags|}
```

Reply:
In the function itself the variable is saved into NON_BARRIER_DIAGNOSTIC_YIELDS. This is how it is saved in two compile functions and passed in from many places using this variable. If we want to change this we should change all of the areas and is not really within the scope of the ticket.

Reply:
> or implement this change in the future when we need to add more flags

Sounds reasonable to me, thanks!

Reply:
yup makes sense @ruby-oujo! We can just leave that for the future. If there is the need to add more flags then we can change that variable to extra_flags.